client/shipping-label/views/index.js

```diff
@@ -1,112 +1,222 @@
 		);
-	};
+	}
+
+	renderRefundLink( label ) {
+		const today = new Date();
+		const thirtyDaysAgo = new Date().setDate( today.getDate() - 30 );
+		if ( ( label.used_date && label.used_date < today.getTime() ) || ( label.created_date && label.created_date < thirtyDaysAgo ) ) {
+			return null;
+		}
+
+		return (
+			<span>
+				<RefundDialog
+					refundDialog={ this.props.shippingLabel.refundDialog }
+					{ ...this.props.shippingLabel }
+					{ ...this.props }
+					{ ...label } />
+				<a href="#" onClick={ () => this.props.labelActions.openRefundDialog( label.label_id ) } ><Gridicon icon="refund" size={ 18 }/> { __( 'Request refund' ) }</a>
+			</span>
+		);
+	}
 
-	const renderLabelActions = ( label, index ) => {
-		if ( label.refunded_time ) {
-			return (
-				<div key={ index }>
-					<dl>
-						<dt>{ __( 'Refund status' ) }</dt>
-						<dd>{ __( 'Refund submitted' ) }</dd>
-
-						<dt>{ __( 'Refund request date' ) }</dt>
-						<dd>{ formatDate( label.refunded_time ) }</dd>
-					</dl>
-				</div>
-			);
+	renderRefund( label ) {
+		if ( ! label.refund ) {
+			return this.renderRefundLink( label );
+		}
+
+		let text = '';
+		let className = '';
+		switch ( label.refund.status ) {
+			case 'pending':
+				className = ( 'wcc-metabox-label-item__refund-pending' );
+				text = __( 'Refund pending' );
+				break;
+			case 'complete':
+				className = ( 'wcc-metabox-label-item__refund-complete' );
+				text = sprintf( __( 'Refunded on %s' ), formatDate( label.refund.refund_date ) );
+				break;
+			case 'rejected':
+				className = ( 'wcc-metabox-label-item__refund-rejected' );
+				text = __( 'Refund rejected' );
+				break;
+			default:
+				return this.renderRefundLink( label );
 		}
 
 		return (
-			<div key={ index }>
-				<dl>
-					<dt>{ __( 'Tracking' ) }</dt>
-					<dd><TrackingLink { ...label }/></dd>
-
-					<dt>{ __( 'Ship via' ) }</dt>
-					<dd>{ label.service_name }</dd>
-
-					<dt>{ __( 'Purchase date' ) }</dt>
-					<dd>{ formatDate( label.created ) }</dd>
-				</dl>
-				<div className="wcc-meta-box-action-buttons">
-					<ReprintDialog
-						reprintDialog={ props.shippingLabel.reprintDialog }
-						{ ...props.shippingLabel }
-						{ ...props }
-						{ ...label } />
-					<Button onClick={ () => props.labelActions.openReprintDialog( label.label_id ) } >
-						{ __( 'Reprint label' ) }
-					</Button>
-
-					<RefundDialog
-						refundDialog={ props.shippingLabel.refundDialog }
-						{ ...props.shippingLabel }
-						{ ...props }
-						{ ...label } />
-					<Button onClick={ () => props.labelActions.openRefundDialog( label.label_id ) } >
-						{ __( 'Refund label' ) }
-					</Button>
-				</div>
+			<span className={ className } ><Gridicon icon="help-outline" size={ 18 }/> { text }</span>
+		);
+	}
+
+	renderReprint( label ) {
+		const todayTime = new Date().getTime();
+		if ( ( label.used_date && label.used_date < todayTime ) || ( label.expiry_date && label.expiry_date < todayTime ) ) {
+			return null;
+		}
```

Comment:
It makes sense to inquire about where this logic should live, given my comments about refund links..